src/server_manager/web_app/app.ts

```diff
@@ -609,21 +608,29 @@ export class App {
 
   private showTransferStats(runningServer: server.Server, serverView: Polymer) {
     const refreshTransferStats = () => {
-      runningServer.getDataUsage().then((stats) => {
-        // Calculate total bytes transferred.
-        let totalBytes = 0;
-        // tslint:disable-next-line:forin
-        for (const accessKeyId in stats.bytesTransferredByUserId) {
-          totalBytes += stats.bytesTransferredByUserId[accessKeyId];
-        }
-        serverView.setServerTransferredData(totalBytes);
-        // tslint:disable-next-line:forin
-        for (const accessKeyId in stats.bytesTransferredByUserId) {
-          const transferredBytes = stats.bytesTransferredByUserId[accessKeyId];
-          const relativeTraffic = totalBytes ? 100 * transferredBytes / totalBytes : 0;
-          serverView.updateAccessKeyRow(accessKeyId, {transferredBytes, relativeTraffic});
-        }
-      });
+      runningServer.getDataUsage().then(
+          (stats) => {
+            // Calculate total bytes transferred.
+            let totalBytes = 0;
+            // tslint:disable-next-line:forin
+            for (const accessKeyId in stats.bytesTransferredByUserId) {
+              totalBytes += stats.bytesTransferredByUserId[accessKeyId];
+            }
+            serverView.setServerTransferredData(totalBytes);
+            // tslint:disable-next-line:forin
+            for (const accessKeyId in stats.bytesTransferredByUserId) {
+              const transferredBytes = stats.bytesTransferredByUserId[accessKeyId];
+              const relativeTraffic = totalBytes ? 100 * transferredBytes / totalBytes : 0;
+              serverView.updateAccessKeyRow(accessKeyId, {transferredBytes, relativeTraffic});
+            }
+          },
+          (e: errors.ServerApiError) => {
+            // Since failure is invisible to users, allow non-network errors percolate to the top
+            // so that a Sentry report will be generated.
+            if (e.response) {
```

Comment:
Shouldn't you be checking the error type here?

Reply:
Hmm, yes. I changed this to check the error type and unless the error is a network error, trigger a Sentry report.